code/client/src/state/modules/wallet/reducers.js

```diff
@@ -146,6 +148,17 @@ const reducer = handleActions(
       ...state,
       provider: action.payload,
     }),
+
+    [walletActions.addKnownAddress]: (state, action) => ({
+      ...state,
+      knownAddresses: {
+        ...state.knownAddresses || {},
+        [action.payload.knownAddressKey]: [
```

Comment:
see above comments on single object, and array of keys v.s. object of [address => {...}]

Reply:
Changed to store known address as object with network and usage details.